File: backend/app/schemas/invitation.py
from pydantic import BaseModel, ConfigDict, Field, EmailStr
from datetime import datetime, timedelta
from typing import Optional
from app.core.config import settings


class InvitationBase(BaseModel):
    """Базовые поля приглашения"""
    family_id: int
    expires_in_days: int = Field(7, ge=1, le=30, description="Срок действия в днях")


class InvitationCreateNewMember(InvitationBase):
    """Приглашение для создания нового члена семьи"""
    invitation_type: str = "new_member"


class InvitationCreateClaimMember(InvitationBase):
    """Приглашение для привязки существующей карточки"""
    member_id: int = Field(..., description="ID карточки родственника")
    invitation_type: str = "claim_member"


class InvitationResponse(BaseModel):
    """Ответ с данными приглашения"""
    id: int
    code: str
    family_id: int
    invitation_type: str
    target_member_id: Optional[int]
    expires_at: datetime
    used_at: Optional[datetime]
    used_by_user_id: Optional[int]
    is_active: bool
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class InvitationCode(BaseModel):
    """Ввод кода приглашения"""
    code: str = Field(..., min_length=4, max_length=20, description="Код приглашения")


class ClaimInvitationResponse(BaseModel):
    """Ответ после активации приглашения"""
    success: bool
    message: str
    family_id: int
    family_name: str
    member_id: Optional[int]
    requires_profile_completion: bool = False

File: frontend/src/api/invitation.ts
import { apiClient } from './client';

export interface Invitation {
  id: number;
  code: string;
  family_id: number;
  invitation_type: 'new_member' | 'claim_member';
  target_member_id: number | null;
  expires_at: string;
  used_at: string | null;
  used_by_user_id: number | null;
  is_active: boolean;
  created_at: string;
}

export interface ClaimInvitationResponse {
  success: boolean;
  message: string;
  family_id: number;
  family_name: string;
  member_id: number;
  requires_profile_completion: boolean;
}

export const invitationAPI = {
  //Создать приглашение для существующего пользователя
  createClaimMemberInvitation: (familyId: number, memberId: number, expiresInDays: number = 7) =>
    apiClient.post<Invitation>('/invitation/create/claim-member', {
      family_id: familyId,
      expires_in_days: expiresInDays,
      member_id: memberId,
      invitation_type: 'claim_member',
    }),
  
  //Активировать приглашение
  claimInvitation: (code: string) =>
    apiClient.post<ClaimInvitationResponse>('/invitation/claim', { code }),
  
  // Получить все приглашения семьи
  getFamilyInvitations: (familyId: number) =>
    apiClient.get<Invitation[]>(`/invitation/family/${familyId}`),
  
  // Отозвать приглашение
  deactivateInvitation: (invitationId: number) =>
    apiClient.delete(`/invitation/${invitationId}`),
};